refactor: fix mypy issues in format_checkers.py

We're in the middle of trying to improve the type hints we use in cve-bin-tool.  I previously filed some issues as found with pyright, this particular set of issues is done using mypy.  We're new to type checking, so it's entirely possible that the types currently in the file are incorrect or incomplete. To test the file yourself, you'll need to install mypy (e.g. `pip install mypy`) and then run it with the filename you want to test.   You may need to also run `mypy --install-types` to get it to work on some files.

Note that I'm filing issues for each file individually to help folk who are looking for fun stuff to do as part of hacktoberfest so you can claim one and not interfere with anyone else working on another file.  If you've already got your 4 commits for hacktoberfest, please leave these issues for someone who's still working on theirs!

Here's a list of issues found:

```
cve_bin_tool/format_checkers.py:86: error: "list" is not subscriptable, use "typing.List" instead
```

This one is probably because python <3.9 needed it to be `List` (capitalized) and imported from the typing module.  You can look at our other code to find an example or look at the mypy cheat sheet https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#built-in-types

Helpful resources:
* [Python typing documentation](https://docs.python.org/3/library/typing.html)
* [mypy cheat sheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)
* If you can't find the answer in the docs, googling error messages will often help!
* [cve-bin-tool's contributor docs](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md) 
* If you've contributed to open source but not this project, you might just want our [checklist for a great pull request](https://github.com/intel/cve-bin-tool/blob/main/CONTRIBUTING.md#checklist-for-a-great-pull-request)

Tiny tips for new contributors:
* cve-bin-tool uses https://www.conventionalcommits.org/ style for commit messages, and we have a test that checks the title of your PR.  A good potential title for this one is in the title of this issue.
* You can make an issue auto close by including a comment "fixes #ISSUENUMBER" in your PR comments where ISSUENUMBER is the actual number of the issue.

--- a/cve_bin_tool/format_checkers.py
+++ b/cve_bin_tool/format_checkers.py
@@ -83,7 +83,7 @@ def update_checker_table(file_path, markdown):
 def update_allowed_words(checkers_array: List[str], file_path: str) -> None:
     """Updates the allow.txt file with the new checkers"""
 
-    checkers_words: list[str] = re.findall(r"[^0-9_]+", "_".join(checkers_array))
+    checkers_words: List[str] = re.findall(r"[^0-9_]+", "_".join(checkers_array))
     with open(file_path) as fileObj:
         words = fileObj.read().strip().splitlines()
 

